usersexercise/src/main/java/com/Services/Implementations/UserServiceImplementation.java

```diff
@@ -0,0 +1,120 @@
+import org.springframework.data.domain.Pageable;
+import java.util.List;
+import java.util.*;
+
+@Service
+public class UserServiceImplementation implements UserService {
+
+    private final UserRepository userrepository;
+
+    public UserServiceImplementation(UserRepository userrepository) {
+        this.userrepository = userrepository;
+    }
+
+
+    @Override
+    public List<UserDto> getUsers(int page, int limit) {
+        List<Users> returnValue;
+        if (page > 0 ) page--;
+        Pageable request = PageRequest.of(page, limit);
+        Page<Users> usersPage = userrepository.findAll(request);
+        returnValue = usersPage.getContent();
+        ArrayList<UserDto> userDtoArrayList = new ArrayList<UserDto>();
+        for(Object x : returnValue){
+            UserDto target = new UserDto();
+            BeanUtils.copyProperties(x, target);
+            userDtoArrayList.add(target);
+        }
+        System.out.println(userDtoArrayList.get(0).getId() + " This is the return Value");
+        return userDtoArrayList;
+    }
+
+    @Override
+    public UserDto createUser(UserDto userdto) {
+        if(userdto.getEmail() == null) throw new ApiRequestException("Most Have an Email");
+        Users newUser = new Users();
+        BeanUtils.copyProperties(userdto, newUser);
+        Users currentUser = userrepository.save(newUser);
+        UserDto returnUser = new UserDto();
+        BeanUtils.copyProperties(currentUser, returnUser);
+        return returnUser;
+    }
+
+    @Override
+    public UserResponse getUser(Long id) {
+        ArrayList<Users> users = (ArrayList<Users>) userrepository.findAll();
+        UserResponse userResponse = new UserResponse();
+        for(int i =0; i < users.size(); i++){
+            if(users.get(i).getId().equals(id)){
+                BeanUtils.copyProperties(users.get(i), userResponse);
+                return userResponse;
+            }
+        }
+
+        throw new ApiRequestException("ID did not match any in the database");
+    }
+
+    @Override
+    public UserResponse getUsersByEmail(String email) {
+        ArrayList<Users> users = (ArrayList<Users>) userrepository.findAll();
+        UserResponse currentUser = new UserResponse();
+        for(int i = 0; i < users.size(); i++){
+            if(users.get(i).getEmail().equals(email)){
+                BeanUtils.copyProperties(users.get(i), currentUser);
+                System.out.println("We are in the user email section");
+                return currentUser;
+            }
+        }
+        throw new ApiRequestException("Email did not match any in the database");
+    }
+
+    @Override
+    public UserResponse updateUser(Users user) {
+        ArrayList<Users> users = (ArrayList<Users>) userrepository.findAll();
+        UserResponse currentUser = new UserResponse();
+        for (int i = 0; i < users.size(); i++) {
+            System.out.println("In the for");
+            if (users.get(i).getId().equals(user.getId())) {
```

Comment:
big problem here, your user request model doesn't include id, so you instead took in a Users object, which would be impossible for a user to do. They don't know their id. Always take in requests from the Controller, and pass Dto to the service